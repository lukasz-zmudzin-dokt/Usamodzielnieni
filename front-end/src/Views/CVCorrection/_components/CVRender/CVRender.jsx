import React from "react";
import { Document, Page, pdfjs } from "react-pdf";
import proxy from 'config/api';
import {Alert} from 'react-bootstrap'

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

const CVRender = ({ url }) => {
  return (
    <div className="cvCorrection__pdfContainer">
      <Document
        className="CVCorrection__pdf"
        file={{
          url: `${proxy.plain}${url}`,
        }}
        error={<Alert variant="danger">Nie udało się załadować CV.</Alert>}
        loading={<Alert variant="info">Ładowanie...</Alert>}
      >
        <Page
          className="pdf_doc_container"
          size="A4"
          pageNumber={1}
          data-testid="pdf"
        />
      </Document>
    </div>
  );
};

export default CVRender;
