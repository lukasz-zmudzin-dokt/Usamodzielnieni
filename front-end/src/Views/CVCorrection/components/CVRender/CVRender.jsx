import React from "react";
import { Document, Page, pdfjs } from "react-pdf";

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

const CVRender = ({ width }) => {
  return (
    <div className="cvCorrection__pdfContainer">
      <Document
        className="CVCorrection__pdf"
        file={{
          url:
            "https://cors-anywhere.herokuapp.com/http://www.africau.edu/images/default/sample.pdf?fbclid=IwAR0q89FFuOH13trzKpQ8OSwOc_A0ANF0QKRwe7L4zrJfCKZi__F8UpgTCNQ",
        }}
      >
        <Page
          className="pdf_doc_container"
          size="A4"
          pageNumber={1}
          width={width}
        />
      </Document>
    </div>
  );
};

export default CVRender;
