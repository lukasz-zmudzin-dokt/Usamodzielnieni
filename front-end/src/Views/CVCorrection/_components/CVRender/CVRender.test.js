import React from "react";
import { render } from "@testing-library/react";
import CVRender from "./CVRender";

describe("CVRender", () => {
  it("should match snapshot", () => {
    const { container } = render(<CVRender />);
    expect(container).toMatchSnapshot();
  });
});
