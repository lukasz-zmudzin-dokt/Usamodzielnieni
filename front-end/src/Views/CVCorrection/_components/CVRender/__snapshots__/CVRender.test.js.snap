// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVRender should match snapshot 1`] = `
<div>
  <div
    class="cvCorrection__pdfContainer"
  >
    <div
      class="react-pdf__Document CVCorrection__pdf"
    >
      <div
        class="react-pdf__message react-pdf__message--loading"
      >
        <div
          class="fade alert alert-info show"
          role="alert"
        >
          ≈Åadowanie...
        </div>
      </div>
    </div>
  </div>
</div>
`;
